travis/script-syntax.py

```diff
@@ -19,14 +16,24 @@ def main():
         if '.git' in parts or 'depends' in parts:
             continue
         for filename in filenames:
-            if not filename.endswith(ext):
+            if not filename.endswith(args.ext):
                 continue
             full_path = os.path.join(cur, filename)
             try:
                 subprocess.check_output(cmd + [full_path])
             except subprocess.CalledProcessError:
                 err = True
+            except IOError:
+                # catch error if not in Travis and Lua/Ruby is not available
```

Comment:
I actually found it pretty useful to run `python travis/all.py` without exceptions being thrown.  But I take your point; changed to instead log a warning once and then exit with 1 (which still isn't an exception, duh).